firefly/app.py

```diff
@@ -47,6 +52,20 @@ def generate_index(self):
             }
         return help_dict
 
+    def generate_showcase(self, **kwargs):
+        functions = [
+                {'name': name, 'path': spec['path'], 'doc': spec['doc'], 'parameters': spec['parameters']}
+                for name, spec in self.generate_function_list().items()
+                ]
```

Comment:
Wouldn't the following line simple enough?

```functions = self.generate_function_list().items()
```

You are calling the function  `generate_function_list` and that is returning a dictionary? Sounds confusing.

Reply:
`generate_function_list` was already there before. It's good that this was pointed out.